packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,264 @@
+    } else if (pagination?.cursorTimestamp) {
+      where.push(sql`timestamp = ${cursorTimestamp}`);
+    } else {
+      where.push(sql`timestamp <= ${cursorTimestamp}`);
+    }
+
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const withoutLimit = sql`
+    SELECT *
+    FROM audit_logs
+    ${whereClause}
+    ORDER BY timestamp DESC, id DESC
+    `;
+    const withLimit = sql`
+    SELECT *
+    FROM audit_logs
+    ${whereClause}
+    ORDER BY timestamp DESC, id DESC
+    LIMIT ${limit}
+    `;
+
+    const query = pagination?.first ? withLimit : withoutLimit;
+    const result = await this.clickHouse.query({
+      query: query,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: string; endDate?: string },
+  ): Promise<{ ok: { url: string } | null } | { error?: { message: string } | null }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      return {
+        error: {
+          message: 'Unauthorized: You are not authorized to perform this action',
+        },
+        ok: null,
+      };
+    }
+
+    const formattedStartDate = filter?.startDate ? new Date(filter.startDate) : undefined;
+    const formattedEndDate = filter?.endDate ? new Date(filter.endDate) : undefined;
+
+    try {
+      const { email } = await this.session.getViewer();
+      const getAllAuditLogs = await this.getPaginatedAuditLogs(organizationSlug, undefined, {
+        startDate: formattedStartDate,
+        endDate: formattedEndDate,
+      });
+      if (!getAllAuditLogs || !getAllAuditLogs.data || getAllAuditLogs.data.length === 0) {
+        return {
+          ok: null,
+          error: {
+            message: 'No audit logs found for the given organization',
+          },
+        };
+      }
+
+      const csvData = await new Promise<string>((resolve, reject) => {
+        stringify(
+          getAllAuditLogs.data,
+          {
+            header: true,
+            columns: {
+              id: 'id',
+              timestamp: 'created_at',
+              event_action: 'event_type',
+              user_id: 'user_id',
+              user_email: 'user_email',
+              metadata: 'metadata',
+            },
+          },
+          (err, output) => {
+            if (err) {
+              reject(err);
+            } else {
+              resolve(output);
+            }
+          },
+        );
+      });
+
+      const s3Storage = this.s3Config[0];
+      const { endpoint, bucket, client } = s3Storage;
+      const cleanStartDate = formattedStartDate?.toISOString().split('T')[0];
+      const cleanEndDate = formattedEndDate?.toISOString().split('T')[0];
```

Comment:
the dates should not be nullable and the `?.toISOString` should be `.toISOString`